drivers/usb/device/usb_dc_nrfx.c

```diff
@@ -262,7 +261,19 @@ struct nrf_usbd_ctx {
 };
 
 
-K_FIFO_DEFINE(work_queue);
+/* FIFO used for queuing up events from ISR. */
+K_FIFO_DEFINE(usbd_evt_fifo);
+
+/* Work queue used for handling the ISR events (i.e. for notifying the USB
+ * device stack, for executing the endpoints callbacks, etc.) out of the ISR
+ * context.
+ * The system work queue cannot be used for this purpose as it might be used in
+ * applications for scheduling USB transfers and this could lead to a deadlock
+ * when the USB device stack would not be notified about certain event because
+ * of a system work queue item waiting for a USB transfer to be finished.
+ */
+static struct k_work_q usbd_work_queue;
+static K_THREAD_STACK_DEFINE(usbd_work_queue_stack, 1024);
```

Comment:
Should the stack size be a Kconfig option? `NRF_USBD_STACK_SIZE`?

Reply:
NRF_USBD_DRIVER_WQUEUE_SIZE?

Reply:
Yes, good point. I'll add such option.

Reply:
Added.